lightning/pkg/importer/table_import.go

```diff
@@ -163,7 +166,7 @@ func (tr *TableImporter) importTable(
 				return false, err
 			}
 
-			checksum, rowIDBase, err := metaMgr.AllocTableRowIDs(ctx, rowIDMax)
+			checksum, rowIDBase, err := metaMgr.AllocTableRowIDs(ctx, requiredRowIDCnt)
```

Comment:
> I forget how the estimated rowIDBase take effect and finally changes the 3 ID allocators

it might be used in kv encoder which will rebase the `panickingAllocator` every time after a row is encoded

> I guess the aloocators should rebase thier values to rowIDBase so different lightning will encode different values.

yes, every instance only calculate how many ID it requires, then call `metaMgr.AllocTableRowIDs` to rebase among all instance

Reply:
> I can only find that allocators are rebased to the real allocator value fetched from downstream TableInfo.

do you mean the code since line 193? that part is used to make sure the final base is >= that defined in table.

Reply:
> then call metaMgr.AllocTableRowIDs to rebase among all instance

I see `AllocTableRowIDs` saves one value into the lightning internal table. But I didn't know where does the `myStartRowID` is used to rebase 3 allocators.

> the code since line 193

In these codes, `myStartRowID` is saved in checkpoint. Does 3 allocators get rebased from the checkpoint?

Reply:
> I see AllocTableRowIDs saves one value into the lightning internal table. But I didn't know where does the myStartRowID is used to rebase 3 allocators.

yes, we only have one value at this phase, it means the the IDs that might be used, when the ID is really used, the panickingAllocator will record it and save to `table` checkpoint where we have 3 values. when post-processing, we will use those 3 values to rebase in tidb side, all parallel instance will do this rebase, and end with the largest ones.

> In these codes, myStartRowID is saved in checkpoint. Does 3 allocators get rebased from the checkpoint?

`myStartRowID` is not saved in checkpoint, it's the range for each file that's saved in `chunk` checkpoint